src/utils/EventWatcher.ts

```diff
@@ -120,22 +149,39 @@ export class EventWatcher {
       }
     );
   }
+}
 
-  getTokenTransfers(logs: Log[], recipient: string) {
-    const ERC20Interface = new utils.Interface(MockERC20Abi.abi);
-    return logs.reduce((logAcc, log) => {
-      try {
-        const parsedLog = ERC20Interface.parseLog(log);
-        if (parsedLog.name === TRANSFER && parsedLog.args.to === recipient) {
-          logAcc.push({
-            token: log.address,
-            amount: parsedLog.args.amount,
-          });
-        }
-        return logAcc;
-      } catch (e) {
-        return logAcc;
+export class RelayEventWatcher extends EventWatcher<RelayOrderReactor> {
+  constructor(provider: BaseProvider, reactorAddress: string) {
+    const reactor = RelayOrderReactor__factory.connect(
+      reactorAddress,
+      provider
+    );
+    super(reactor);
+  }
+
+  async getFillLogs(fromBlock: number, toBlock?: number): Promise<FillEvent[]> {
```

Comment:
looks like we are using getFillInfo in the service. Is there a difference in implementation form that and getFillLogs?

Reply:
reference
```
export declare type TokenTransfer = {
    token: string;
    amount: BigNumber;
};
export interface FillData {
    orderHash: string;
    filler: string;
    nonce: BigNumber;
    swapper: string;
}
export interface FillInfo extends FillData {
    blockNumber: number;
    txHash: string;
    inputs: TokenTransfer[];
    outputs: TokenTransfer[];
}
/**
 * Helper for watching events
 */
export declare class EventWatcher {
    private reactor;
    constructor(provider: BaseProvider, reactorAddress: string);
    getFillLogs(fromBlock: number, toBlock?: number): Promise<FillEvent[]>;
    getFillEvents(fromBlock: number, toBlock?: number): Promise<FillData[]>;
    getFillInfo(fromBlock: number, toBlock?: number): Promise<FillInfo[]>;
    onFill(callback: (fillData: FillData, event: Event) => void): void;
    getTokenTransfers(logs: Log[], recipient: string): {
        token: string;
        amount: BigNumber;
    }[];
}

```

Reply:
getFillLogs is a function called by getFillInfo, it gets the raw logs on chain and fill info adds order specific data